### DISCUSSION TITLE
Does MVT style spec support inward or outward blur for line and polygon?

### DISCUSSION THREAD

**Discussion by zhangyiatmicrosoft** [maplibre/maplibre-gl-js#2385] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/37553549/231628439-f8e711d3-1d46-4148-b745-ae6c640f26f0.png)

Hello community, please see picture above. Does MVT style support inward or outward blur for line and polygon?
This is useful for rendering coast lines or lakes or similar objects.

If not MVT style, can this effect be archived by any other way? Thanks

**Comment by IvanSanchez** [comment, marked as answer]
You're not looking for a *blur*; rather, you're looking for a *gradient*.

One way of achieving this is using `line-pattern`, with the pattern being an image one pixel wide, and as high as the width of your line. The image should be a gradient fully opaque at one end, and fully transparent at the other.

**Comment by zhangyiatmicrosoft** [comment]
Thanks!
